aws/subscriber.go

```diff
@@ -51,15 +51,33 @@ type awsSubscriber struct {
 	wg sync.WaitGroup
 }
 
-func (s *awsSubscriber) Start(ctx context.Context, filter map[string]string) (<-chan pubsub.Message, <-chan error) {
+func (s *awsSubscriber) Start(ctx context.Context, opts ...pubsub.Option) (<-chan pubsub.Message, <-chan error) {
+	// Default Options for AWS subscriber
+	subscriberOptions := &pubsub.Options{
+		VisibilityTimeout: 30 * time.Second,
+		RetentionPeriod:   345600 * time.Second,
+	}
+	for _, opt := range opts {
+		opt(subscriberOptions)
+	}
 	channel := make(chan pubsub.Message)
 	errChannel := make(chan error)
 	go func() {
 		defer close(channel)
-		if fErr := s.ensureFilterPolicy(filter); fErr != nil {
+		if fErr := s.ensureFilterPolicy(subscriberOptions.Filter); fErr != nil {
```

Comment:
it'd be nice to be able to do these three things in a single call instead of 3 independent calls... there's a stricter AWS rate limit on these calls so we should try to minimize them.

Reply:
We wouldn't be able to do all 3 calls together because the filter goes through sns not sqs, and we need the sqs call for (retention and visibility). But we can do the 2 (visibility and retention) together.